### DISCUSSION TITLE
Swarm dial returns ok even if peer is not accessible

### DISCUSSION THREAD

**Discussion by haider-rs** [libp2p/rust-libp2p#2844] [Q&A] [answered]
Hey i tried to dial a random multiaddr which is not accessible. but if I write it in swam dial it returns Ok 

**rust code**:
```
use libp2p::{Swarm, core::transport::dummy::DummyTransport, swarm::DummyBehaviour, PeerId, Transport, Multiaddr};

fn main() {
    let mut swarm = Swarm::new(
        DummyTransport::new().boxed(),
        DummyBehaviour::default(),
        PeerId::random(),
    );

    // Dial an unknown peer.
    if let Ok(_) = swarm.dial("/ip4/100.24.100.100/tcp/1234".parse::<Multiaddr>().unwrap()){
        println!("dialed");
    }else{
        println!("error dialing");
    }
}

```

Output:
`dialed`

libp2p-lookup:
`libp2p-lookup direct --address /ip4/100.24.100.100/tcp/1234`
Output:
`[2022-08-24T12:43:43Z ERROR libp2p_lookup] Lookup failed: Timeout.`

Is this the expected behavior? 
If so can I please know the responsibility of swarm dialing?
and is there any way I can check connectivity using some other function? 
I know Ping but it returns an event and I want to check this before starting network loop.

Thanks

**Comment by thomaseizinger** [comment, marked as answer]
`dial` is asynchronous and only some pre-conditions are checked synchronously. You need to `poll` the Swarm (via `next` for example) and have it make progress which will eventually yield an event that informs you whether the dial was successful.

**Reply by haider-rs** [reply]
thanks
